### PR TITLE
refactor: move some functions to winfloat.c

### PR DESCRIPTION
**Pull request by glepnir**
window.c is too long. moved to winfloat.c (like vim popupwin.c) It is convenient to add any floating window related functions later.

### CODE REVIEW

**Comment by glepnir** [comment]
@zeertzjq  could you PTAL? seems like auto include not works where i missing. must `include floatwin.h.generated.h` otherwise I got many no previous prototype of function.

File: src/nvim/floatwin.c [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+{
+  for (win_T *wp = lastwin; wp && wp->w_floating; wp = wp->w_prev) {
+    win_config_float(wp, wp->w_float_config);
+  }
+}
+
+/// Return true if "win" is floating window in the current tab page.
+///
+/// @param  win  window to check
+bool win_valid_floating(const win_T *win)
```
Comment (justinmk) [inline src/nvim/floatwin.c:265]: `win_float` is probably a good prefix to use for the function names.

I guess the file should be named `winfloat.c/h` so it is easily discoverable next to `window.c/h`

Comment (justinmk) [inline src/nvim/floatwin.c:265]:
```suggestion
bool win_float_valid(const win_T *win)
```

Comment (glepnir) [reply src/nvim/floatwin.c]: sure

File: src/nvim/winfloat.c [hunk 2]
```diff
@@ -0,0 +1,278 @@
+// This is an open source non-commercial project. Dear PVS-Studio, please check
+// it. PVS-Studio Static Code Analyzer for C, C++ and C#: http://www.viva64.com
+
```
Comment (zeertzjq) [inline src/nvim/winfloat.c:3]: PVS has been removed
```suggestion
```

**Review by justinmk — APPROVED** [review]

File: src/nvim/winfloat.c [hunk 3]
```diff
@@ -0,0 +1,275 @@
...
+#include "nvim/grid.h"
+#include "nvim/mouse.h"
+#include "nvim/move.h"
+#include "nvim/option.h"
+#include "nvim/optionstr.h"
+#include "nvim/ui.h"
+#include "nvim/vim.h"
+#include "nvim/window.h"
+#include "nvim/winfloat.h"
+
```
Comment (zeertzjq) [inline src/nvim/winfloat.c:17]: Include `winfloat.c.generated.h`

Comment (zeertzjq) [inline src/nvim/winfloat.c:17]:
```suggestion

#ifdef INCLUDE_GENERATED_DECLARATIONS
# include "winfloat.c.generated.h"
#endif

```

**Pull request merged**